File: hw3/files/polydyn.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cmath>
#include "polydyn.h"
using namespace std;
using namespace hw5;


polydyn::polydyn(rectangle& rec){
		x = new double[4];
		y = new double[4];
		areaa = rec.area();
		perimeterr = rec.perimeter();
		x[0] = rec.get_x();
		y[0] = rec.get_y();
		x[1] = rec.get_x()+rec.get_width();
		y[1] = rec.get_y();
		x[2] = rec.get_x()+rec.get_width();
		y[2] = rec.get_y()+rec.get_height();
		x[3] = rec.get_x();
		y[3] = rec.get_y()+rec.get_height();
		size = 4;
}
polydyn::polydyn(circle& cir){
		areaa = cir.area();
		perimeterr = cir.perimeter();
		x = new double [200];
		y = new double [200];
		size = 100;
		double i = 3.6;
		int j = 0;
		while(i <= 360){
			i +=3.6*j*(3.14/180);
			x[j] = cir.get_radius()*(cos(i))+cir.get_x();

			y[j] = cir.get_radius()*(sin(i))+cir.get_y();
			
			j++;
		}
}
polydyn::polydyn(triangle& tri){
	    areaa = tri.area();
		perimeterr = tri.perimeter();
		x = new double [3];
		y = new double [3];
		size = 3;
		x[0] = tri.get_x();
		y[0] =tri.get_y();
		x[1] = tri.get_x2();
		y[1] = tri.get_y2();
		x[2] =tri.get_x3();
		y[2] =tri.get_y3();
}
void polydyn::draw(ofstream& file)const{
	file<<"<polygon points=\"";
	for(int i = 0;i <size;i++){
		file<<x[i]<<","<<y[i];
		if(i == size-1)
			file<<"\"";
		else
			file<<" ";
	}
	file<<" style=\"fill:lime;stroke:purple;stroke-width:2;fill-rule:nonzero;\" />\n";


}
polydyn& polydyn::operator++(){
	for(int i = 0 ; i < size ; i++){
		x[i] = x[i]+1;
		y[i] = y[i]+1;
	}
	return *this;
};

polydyn& polydyn::operator++(int temp){
	polydyn& poly = *this;
	for(int i = 0 ; i < size ; i++){
			x[i] = x[i]+1;
			y[i] = y[i]+1;
		}
		return poly;

};
polydyn& polydyn::operator--(){
	for(int i = 0 ; i < size ; i++){
		x[i] = x[i]-1;
		y[i] = y[i]-1;
	}
	return *this;
};
polydyn& polydyn::operator--(int temp){
	polydyn& poly = *this;
	for(int i = 0 ; i < size ; i++){
			x[i] = x[i]-1;
			y[i] = y[i]-1;
		}
		return poly;
};
polydyn::~polydyn(){
	delete []x;
	delete []y;
	x = NULL;
	y = NULL;
}
polydyn::polydyn(const polydyn& obj){
	size = obj.get_size();
	x = new double[size];
	y = new double[size];
	copy(x,x+size,obj.x);
	copy(y,y+size,obj.y);
	
	areaa = obj.areaa;
	perimeterr = obj.perimeterr;

}

polydyn& polydyn::operator=(const polydyn& poly){
	double* newdata_x = new double[poly.get_size()];
	double* newdata_y = new double[poly.get_size()];
	copy(poly.x, poly.x+poly.get_size() , newdata_x);
	copy(poly.y, poly.y+poly.get_size() , newdata_y);
	delete[] x;
	delete[] y;
	x = newdata_x;
	y = newdata_y;
	areaa = poly.areaa;
	perimeterr = poly.perimeterr;
	size = poly.get_size();
	return *this;	
}

File: hw5/hw7/PolyDyn.java
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;
import java.util.ArrayList;
import java.lang.Math; 



public class PolyDyn extends Polygon{
	private point2D[] points;
    public PolyDyn(){}
	public PolyDyn(Circle cir){
		size = 200;
		points = new point2D[size];
		area = cir.area();
		perimeter = cir.perimeter();
		double i = 3.6;
		int j = 0;

	 points = new point2D[size];
	 point2D point ;
	while(i <= 360){
		i +=3.6*j*(3.14/180);
		double a = cir.get_radius()*(Math.cos(Math.toRadians(i)))+cir.get_x();
		double b = cir.get_radius()*(Math.sin(Math.toRadians(i)))+cir.get_y();
		
		point = new point2D(a,b);
		
		points[j]=point;
		j++;}
		

	}
	public PolyDyn(Rectangle rec){
		area = rec.area();
		perimeter = rec.perimeter();
		size = 4;
		points = new point2D[size];
		point2D point = new point2D(rec.get_x(),rec.get_y());
		points[0] = point;
		point = new point2D(rec.get_x()+rec.get_width(),rec.get_y());
		points[1] = point;
		point = new point2D(rec.get_x()+rec.get_width(),rec.get_y()+rec.get_height());
		points[2] = point;
		point = new point2D(rec.get_x(),rec.get_y()+rec.get_height());
		points[3] = point;
		

	}
	public PolyDyn(Triangle tri){
		size = 3;
		area = tri.area();
		perimeter = tri.perimeter();
		points = new point2D[size];
		point2D point = new point2D(tri.get_x(),tri.get_y());
		points[0] = point;
		point2D point2 = new point2D(tri.get_x2(),tri.get_y2());
		points[1] = point2;

		point2D point3 = new point2D(tri.get_x3(),tri.get_y3());
		points[2] = point3;
	}
	public void draw(Graphics obj){
		int x[] = new int[size];
		int y[] = new int[size];
		for(int i = 0 ; i < size; i++){
			x[i] = (int)points[i].get_x();
			y[i] = (int)points[i].get_y();
		}
		obj.setColor(Color.red);
	 	obj.drawPolygon(x,y,size);
	 	obj.setColor(Color.pink);
	 	obj.fillPolygon(x,y,size);
	}
	public void paintComponent(Graphics obj){
		super.paintComponent(obj);
	 	draw(obj);
	}
	public void set(double a , double b){}
	public  double get_x(){return 0;}
	public  double get_y(){return 0;}
	public double area(){return area;}
	public double perimeter(){return perimeter;}
	public int compareTo(shape temp){return 0;}

	public void post_increment(){
		for(int i = 0 ; i < size ; i++){
			points[i].post_increment();
		}

	}
	public void pre_increment(){
		for(int i = 0 ; i < size ; i++){
			points[i].pre_increment();
		}
	}
	public void post_decrement(){
		for(int i = 0 ; i < size ; i++){
			points[i].post_decrement();
		}
	}
	public void pre_decrement(){
		for(int i = 0 ; i < size ; i++){
			points[i].pre_decrement();
		}

	}

}